docs/querying/tips-good-queries.md

```diff
@@ -0,0 +1,198 @@
+
+#### Use approximation
+
+When possible, design your SQL queries in such a way that they match the rules for TopN approximation, so that Druid enables TopN by default. For Druid to automatically optimize for TopN, your SQL query must include the following:
+
+- GROUP BY on one dimension, and
+- ORDER BY on one aggregate.
+
+ See [TopN queries](./topnquery.md) for more information.
+
+Note that TopN queries are approximate in that each data process ranks its top K results and only returns those top K results to the Broker.
+
+You can follow the tutorial [Using TopN approximation in Druid queries](https://github.com/apache/druid/blob/master/examples/quickstart/jupyter-notebooks/notebooks/03-query/02-approxRanking.ipynb) to work through some examples with approximation turned on and off. The tutorial [Get to know query view](../tutorials/tutorial-sql-query-view.md) demonstrates running aggregate queries in the Druid console.
+
+### Manually tune your queries
+
+This section outlines techniques you can use to improve your query accuracy and performance.
+
+#### Query one table at a time
+
+Query a single table at a time to minimize the load on the Druid processor.
+
+#### Select specific columns
+
+Only select the columns needed for the query instead of retrieving all columns from the table. This reduces the amount of data retrieved from the database, which improves query performance.
+
+#### Use filters
+
+Use filters, for example the WHERE clause, and filter on time. Try to minimize the use of non-equal filters, because they're very resource-intensive.
+
+The following example query filters on `__time` and `product`:
+
+```
+SELECT
+  FLOOR(__time to day),
+  product,
+  sum(quantity * price) as revenue
+FROM "orders"
+WHERE
+  __time > '2023-08-20' and product = 'product 1'
+GROUP BY 1, 2
+```
+
+The following example uses a wildcard filter on the `diffUrl` column:
+
+```
+SELECT * from Wikipedia
+WHERE diffUrl LIKE 'https://en.wikipedia%'
+AND __time BETWEEN '2016-06-27 00:00' AND '2016-06-27 12:00'
+```
+
+#### Shorten your queries
+
+Make your queries shorter where possible&mdash;Druid processes shorter queries faster. You might also be able to divide a single query into multiple queries.
+
+For example, the following query aggregates over multiple datasources using UNION ALL:
+
+```
+SELECT id, SUM(revenue) FROM
+   (SELECT id, revenue from datasource_1
+UNION ALL
+  SELECT id, revenue FROM datasource_2)
+...
+UNION ALL
+   SELECT id, revenue FROM datasource_n)
+GROUP BY id
+```
+
+To simply this query, you could split it into several queries and then aggregate the results. For example:
+
+```
+SELECT id, SUM(revenue) FROM datasource_1
+
+SELECT id, SUM(revenue) FROM datasource_2
+...
+SELECT id, SUM(revenue) FROM datasource_n
+```
+
+#### Minimize or remove subqueries
+
+Consider whether you can pre-compute a subquery task and store it as a join or make it a part of the datasource.
```

Comment:
Further explanation, links to relevant docs, and/or examples would be helpful here

Reply:
Added a couple of links to more information and examples.